File: packages/dd-trace/test/llmobs/parity/scenarios/openai/responses-tool-call.py
import os

from openai import OpenAI

from common import finish, start


start("openai")
client = OpenAI(api_key=os.environ["OPENAI_API_KEY"], base_url=f'{os.environ["PROVIDER_BASE_URL"]}/v1')
client.responses.create(
    model="gpt-4o-mini",
    input="What is the weather in NYC?",
    tools=[
        {
            "type": "function",
            "name": "get_weather",
            "description": "Get current weather.",
            "parameters": {
                "type": "object",
                "properties": {"city": {"type": "string"}},
                "required": ["city"],
            },
        }
    ],
)
finish()

File: packages/dd-trace/test/llmobs/parity/scenarios/openai/responses-tool-call.js
'use strict'

const { finish, loadTracer, loadVersionedModule } = require('../common')

async function main () {
  const tracer = loadTracer('openai')
  const OpenAI = loadVersionedModule('openai')
  const client = new OpenAI({ apiKey: process.env.OPENAI_API_KEY, baseURL: `${process.env.PROVIDER_BASE_URL}/v1` })
  await client.responses.create({
    model: 'gpt-4o-mini',
    input: 'What is the weather in NYC?',
    tools: [{
      type: 'function',
      name: 'get_weather',
      description: 'Get current weather.',
      parameters: {
        type: 'object',
        properties: { city: { type: 'string' } },
        required: ['city'],
      },
    }],
  })
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack ?? error}\n`)
  process.exitCode = 1
})
